psi4/driver/qcdb/align.py

```diff
@@ -26,148 +26,10 @@
-            algeom = algeom + self.shift
-            if self.mirror:
-                algeom[:, 1] *= -1.
-        else:
-            if self.mirror:
-                algeom[:, 1] *= -1.
-            algeom = algeom - self.shift
-            algeom = algeom.dot(self.rotation)
-        algeom = algeom[self.atommap, :]
-        # mirror-wise, rsm/msr == rms/msr
-
-        return algeom
-
-    def align_atoms(self, ats):
-        """suitable for masses, symbols, Zs, etc."""
-
-        return ats[self.atommap]
-
-    def align_vector(self, vec):
-        """suitable for vector attached to molecule"""
-
-        # sensible? TODO
-        #alvec = np.copy(vec)
-        #if self.mirror:
-        #    alvec[:, 1] *= -1
-        return vec.dot(self.rotation)
-
-    def align_gradient(self, grad):
-        """suitable for vector system attached to atoms"""
-
-        # sensible? TODO
-        #algrad = np.copy(grad)
-        #if self.mirror:
-        #    algrad[:, 1] *= -1
-        algrad = grad.dot(self.rotation)
-        algrad = algrad[self.atommap]
-
-        return algrad
-
-    def align_hessian(self, hess):
-        blocked_hess = qcdb.util.blockwise_expand(hess, (3, 3), False)
-        alhess = np.zeros_like(blocked_hess)
-
-        nat = blocked_hess.shape[0]
-        for iat in range(nat):
-            for jat in range(nat):
-                alhess[iat, jat] = (self.rotation.T).dot(blocked_hess[iat, jat].dot(self.rotation))
-
-        alhess = alhess[np.ix_(self.atommap, self.atommap)]
-
-        alhess = qcdb.util.blockwise_contract(alhess)
-        return alhess
-
-    def align_system(self, geom, mass, elem, elez, uniq, reverse=False):
-        """For AlignmentRecipe `ar`, apply its translation, rotation, and atom map."""
-
-        nugeom = self.align_coordinates(geom, reverse=reverse)
-        numass = self.align_atoms(mass)
-        nuelem = self.align_atoms(elem)
-        nuelez = self.align_atoms(elez)
-        nuuniq = self.align_atoms(uniq)
-
-        return nugeom, numass, nuelem, nuelez, nuuniq
-
-    def align_mini_system(self, geom, uniq, reverse=False):
-        """For AlignmentRecipe `ar`, apply its translation, rotation, and atom map."""
-
-        nugeom = self.align_coordinates(geom, reverse=reverse)
-        nuuniq = self.align_atoms(uniq)
-
-        return nugeom, nuuniq
-
-
-def _nre(Z, geom):
-    """Nuclear repulsion energy"""
-
-    nre = 0.
-    for at1 in range(geom.shape[0]):
-        for at2 in range(at1):
-            dist = np.linalg.norm(geom[at1] - geom[at2])
-            nre += Z[at1] * Z[at2] / dist
-    return nre
-
-
-def _pseudo_nre(Zhash, geom):
-    """Pseudo nuclear repulsion energy where non-physical Z contrived from `Zhash`."""
-
-    Zidx = list(set(sorted(Zhash)))
-    pZ = [Zidx.index(z) for z in Zhash]
-    return _nre(pZ, geom)
-
-
-#         1         2         3         4         5         6         7         8         9         10
-
 
 def B787(cgeom,
```

Comment:
Didn't even realise we had this module available. Great. Why is this fn called "B787" - anything to do with the airplane?

Reply:
Yep, it can be handy. Now lives [here](https://github.com/MolSSI/QCElemental/blob/master/qcelemental/molutil/align.py#L33) but you'll probably want to call it through `psi4.core.Molecule.B787`, `qcdb.Molecule.B787`, `qcel.models.Molecule.align` (I was persuaded to rename the last on the findability grounds). Its definitely good through coupled cluster sized mols. Yes, the airplane -- the function's nickname in the lab is "dreamaligner."